### ISSUE TITLE
Exposing getToken();

### ISSUE THREAD

**Issue by schuetta92** [paulvanbladel/aurelia-auth#49]
I was thinking about exposing the getToken() method to be an available function, but was wondering if there is a reason it's not, if there is security reasons as to why it's not?

I am trying to download a file from my server and in order to do that I have to link it in the href, but I also need to verify that the user is authenticated so I was going to put the token in the query string? 

so when the user clicks on 
```
<a href="/server/downloads/.txt" id="wineMenu">Wine Menu</a>
```

I want it to download the response from the following call, which needs the token in the auth middleware. 
```
router.get('/:file(*)', auth, function(req, res){
  var file = 'wine-menu.txt'
    , path = __dirname + '/files/' + file;
  res.download(path);
});
```

**Comment by paulvanbladel** [comment]
Indeed, using the query string for sending the token to the server is a real anti-pattern.
Check the OWASP top 10 for more info. Using a query string is making things too overt, people can copy/paste urls and use your identity (at least as long as the token is not expired).
Normally, the token is automatically sent for every request to the server.

**Comment by schuetta92** [comment]
Okay that's kind of what I was thinking but it seemed like an easy solution. 

Is there a way to do what I am looking to do above then? Since it doesn't send the token with the href request? If I try and use the httpclient to get the url the request doesn't download.

The response from the Http client get call response looks like this 

```
{
headers: 
    {
    headers:
       {
        Accept-Ranges: "bytes",
        Cache-Control: "public, max-age=0",
        Connection: "Keep-Alive",
        Content-Disposition: "attachment; filename="wine-menu.txt"",
        Content-Encoding: "gzip",
        Content-Type: "text/plain; charset=UTF-8",
        Date: "Fri, 13 Nov 2015 08:13:14 GMT",
        ETag: "W/"d-150ffd43dd8""Keep-Alive: "timeout=5, max=43",
        Last-Modified: "Fri, 13 Nov 2015 07:51:51 GMT",
        Transfer-Encoding: "chunked",
        Vary: "Accept-Encoding",
        X-Powered-By: "Express"
       }
    },
    isSuccess: true,
    mimeType: "text/plain",
    requestMessage: HttpRequestMessage,
    response: "Hello world!!",
    responseType: "txt",
    reviver: undefinedstatusCode: 200,
    statusText: "OK"
}

```

**Comment by paulvanbladel** [comment]
I found following in aurelia docs
If you want to navigate to the generated URL, use router.navigateToRoute('userDetail', { id: 123 }).

If you simply want to render an anchor in your view, you can use the route-href custom attribute.

<a route-href="route: userDetail; params.bind: { id: user.id }">${user.name}</a>
Extending HTML
